static/js/protein/chart.ts

```diff
@@ -47,116 +46,558 @@ export interface ProteinPropDataStrType {
+      'Pancreas':'peachpuff', 'Thyroid':'lightcoral', 'Whole Blood':'firebrick'
+    }
+    // returns corresponding color for tissue
+    function circleColor(d: string) {
+      var name = d;
+      return(var_color[name])
+    };
+
+    var svg = d3.selectAll("#variant-gene-association-chart")
+      .append("svg")
+      .attr("width", width + MARGIN.left + MARGIN.right)
+      .attr("height", height + MARGIN.top + MARGIN.bottom)
+      .append("g")
+          .attr("transform",
+                "translate(" + MARGIN.left + "," + MARGIN.top + ")");
+    // plots the axes
+    var x = d3.scaleLinear()
+        .domain([d3.min(reformattedData, d => d.lower), d3.max(reformattedData, d => d.upper)])
+        .range([0, width]);
+    svg.append("g")
+        .attr("transform", "translate(0," + height + ")")
+        .call(d3.axisBottom(x))
+    var y = d3.scaleBand()
+        .range([ 0, height ])
+        .domain(reformattedData.map(function(d) { return d.id; }))
+        .padding(1);
+    svg.append("g")
+        .call(d3.axisLeft(y))
+    // adds the dots and error bars 
+    svg.selectAll("myline")
+      .data(reformattedData)
+      .enter()
+      .append("line")
+        .attr("x1", function(d) { return x(d.upper); })
+        .attr("x2", function(d) { return x(d.lower); })
+        .attr("y1", function(d) { return y(d.id); })
+        .attr("y2", function(d) { return y(d.id); })
+        .attr("stroke", "grey")
+        .attr("stroke-width", "1px")
+    svg.selectAll("mycircle")
+        .data(reformattedData)
+        .enter()
+        .append("circle")
+          .attr("cx", function(d) { return x(d.value); })
+          .attr("cy", function(d) { return y(d.id); })
+          .attr("r", "6")
+          .style("fill", function (d) {return circleColor(d.name)})
+    svg.append("circle").attr("cx",450).attr("cy",10).attr("r", 6).style("fill", "peachpuff")
+    svg.append("circle").attr("cx",450).attr("cy",30).attr("r", 6).style("fill", "lightcoral")
+    svg.append("circle").attr("cx",450).attr("cy",50).attr("r", 6).style("fill", "firebrick")
+    svg.append("text").attr("x", 470).attr("y", 10).text("Pancreas").style("font-size", "15px").attr("alignment-baseline","middle")
+    svg.append("text").attr("x", 470).attr("y", 30).text("Thyroid").style("font-size", "15px").attr("alignment-baseline","middle")
+    svg.append("text").attr("x", 470).attr("y", 50).text("Whole Blood").style("font-size", "15px").attr("alignment-baseline","middle")
+
+
+
+  }
+  
+ 
+ export function drawVarTypeAssocChart(
+   id: string,
+   data : {name: string; value: number}[]
+ ) : void {
+      // formats variant name 
+      function formatVariant(d: string) {
+        d = d.substring(32);
+        return d;
+      }
+      const height = 300 - MARGIN.top - MARGIN.bottom;
+      const width = 600 - MARGIN.left - MARGIN.right;
+      data.sort((a, b) => {
+        return b.value - a.value;
+      });
+      const size = data.map(x => { return x.name;});
```

Comment:
`size` might not be the best variable name for a list of variant names?

Reply:
Renamed it !